packages/core/src/domain/user/user.ts

```diff
@@ -30,3 +33,41 @@ export function checkUser(newUser: User): boolean {
   }
   return isValid
 }
+
+export function getAnonymousIdFromStorage(): string | undefined {
+  let matches = /aid=([\w-]+)/.exec(document.cookie)
+  if (matches) {
+    return matches[1]
+  }
+
+  matches = /aid=(\w+)/.exec(localStorage.getItem(SESSION_STORE_KEY) ?? '')
+  if (matches) {
+    return matches[1]
+  }
+}
+
+export function setAnonymousIdInStorage(sessionStoreStrategyType: string, anonymousId: string) {
+  let sessionString =
+    sessionStoreStrategyType === 'Cookie' ? getCookie(SESSION_STORE_KEY) : localStorage.getItem(SESSION_STORE_KEY)
+  sessionString = isValidSessionString(sessionString) ? `${sessionString}&aid=${anonymousId}` : `aid=${anonymousId}`
+
+  if (sessionStoreStrategyType === 'Cookie') {
+    setCookie(SESSION_STORE_KEY, sessionString)
+  } else {
+    localStorage.setItem(SESSION_STORE_KEY, sessionString)
+  }
+}
+
+export function retrieveAnonymousId(sessionStoreStrategyType: string): string {
+  let anonymousId = getAnonymousIdFromStorage()
+
+  if (!anonymousId) {
+    anonymousId = generateAnonymousId()
+    setAnonymousIdInStorage(sessionStoreStrategyType, anonymousId)
+  }
+  return anonymousId
+}
```

Comment:
**💬 suggestion:** ‏I know the current SessionState/SessionStore etc. is a bit hard to grasp, but I really think we should keep a single way of writing and reading the session cookie. Let's look at this together?

Reply:
Yes, thank you! I agree on keeping a single way. But I don't think we want to tie this anonymous id logic with session state logics, as the latter one needs to be refreshed oftenly. I tried to do it directly from the `get/set` cookie functions which did not work.